std/format.d

```diff
@@ -3040,19 +3040,46 @@ if (isInputRange!T)
             static if (is(StringTypeOf!T))
             {
                 auto s = val[0 .. f.precision < $ ? f.precision : $];
+
+                size_t width;
+                if (f.width > 0)
+                {
+                    // strings that are fully made of ASCII characters
+                    // can be aligned w/o graphemeStride
+                    bool onlyAscii = true;
+                    for (size_t i; i < s.length; i++)
+                    {
+                        width += s[i] >= 30;
```

Comment:
I disagree with this change.  The characters under ASCII 0x20 (32) are control characters, and can have effects on formatting. For example, `\n` is ASCII 0x0A (13) (on Posix anyway, on Windows you also get 0x0D with it), and has the side-effect of breaking the line completely. So what "width" should it have?  Similarly, `\t` is ASCII 0x09, and has the side-effect of changing the current horizontal position of the output differently depending on how the target terminal interprets it.  There are a few others that have similar side-effects, not to mention escape sequences prefixed by ASCII 0x1B, that changes the interpretation of an arbitrary number of following characters, again depending on how the terminal interprets it.

IMO it's outside the charter of `std.format` to interpret these control characters, so asking it to format these characters into a particular width is basically UB.  As such, the implementation should not sacrifice speed/complexity to treat them specially.  It's not *wrong* per se to treat them as if they have zero width, but I don't think we should go out of our way just to do that.

Reply:
Note furthermore that we cannot just ignore them, because the user *may* have inserted them specifically because they wanted the side-effects associated with them. E.g., we regularly write `\n` to introduce line breaks into the output, and conceivably, someone could call `format("%s", obj)` and `obj.toString` might return a multi-line string.   But what should be done in the case `format("%.2s", "\t")`?  Or indeed, `format("%.1s", "\nX")`?  Since these cases have no sane answer, my inclination is to call them UB and just not handle them at all in the code.

Reply:
(Also, the code as-is is wrong, because ASCII 30 and 31 are also control characters, but the current if-condition would treat them as having width 1. The correct condition should be `>= 32`.)

Reply:
Ok removed. Since you're there you can re-label for merging.